clang/lib/Sema/SemaOverload.cpp

```diff
@@ -1595,8 +1606,21 @@ static bool IsOverloadOrOverrideImpl(Sema &SemaRef, FunctionDecl *New,
 
         // Allow overloading of functions with same signature and different CUDA
         // target attributes.
-        if (NewTarget != OldTarget)
+        if (NewTarget != OldTarget) {
+          // Special case: non-constexpr function is allowed to override
+          // constexpr virtual function
+          const auto *OldMethod = dyn_cast<CXXMethodDecl>(Old);
+          const auto *NewMethod = dyn_cast<CXXMethodDecl>(New);
```

Comment:
OldMethod/NewMethod should still be in scoped and don't need to be redeclared

Reply:
will do